### PR TITLE
Support using subsequences in data loader

### PR DESCRIPTION
**Pull request by PhysSong**
This is required to perform single-step training, as documented in section A.3 in the paper.
`num_frames` and `subseq_len` probably require better names, so please let me know if you have any ideas.

### CODE REVIEW

**Review by yycho0108 — APPROVED** [review]
LGTM. Do we need to shuffle the dataset to prevent the correlations between sub-sequences affecting the training? (or perhaps `torch.util.DataLoader` might already take care of this).

**Comment by PhysSong** [comment]
> Do we need to shuffle the dataset to prevent the correlations between sub-sequences affecting the training? (or perhaps `torch.util.DataLoader` might already take care of this).

`DataLoader` has an option to enable shuffling, but it's turned off by default. It seems like the authors have shuffling enabled implicitly by `DistributedSampler`. Should we turn on that as well?

**Comment by yycho0108** [comment]
> > Do we need to shuffle the dataset to prevent the correlations between sub-sequences affecting the training? (or perhaps `torch.util.DataLoader` might already take care of this).
> 
> `DataLoader` has an option to enable shuffling, but it's turned off by default. It seems like the authors have shuffling enabled implicitly by `DistributedSampler`. Should we turn on that as well?

If we're using sub-sequences, I think we should enable shuffling.
